Cabal/Distribution/Simple/Build.hs

```diff
@@ -206,9 +206,10 @@ buildComponent verbosity numJobs pkg_descr lbi suffixes
         -- Register the library in-place, so exes can depend
         -- on internally defined libraries.
         pwd <- getCurrentDirectory
+        abi_hash <- abiHash verbosity pkg_descr distPref lbi lib clbi
         let -- The in place registration uses the "-inplace" suffix, not an ABI hash
             installedPkgInfo = inplaceInstalledPackageInfo pwd distPref pkg_descr
-                                                           (mkAbiHash "") lib' lbi clbi
+                                                           abi_hash lib' lbi clbi
```

Comment:
Wait, is that really necessary? That means computing the hash on every build. Surely using an empty abi also works here. The case where this comes up is other libs within the same package convenience/backpack and their abi depends. But that should also work with empty ABIs (and if not, that should be fixed). Remember, we have compilers for which we have no sensible hash (and we currently use empty for those).